### ISSUE TITLE
Strange Issues with Image Classification Transfer Learning Sample 

### ISSUE THREAD

**Issue by reneschulte** [dotnet/machinelearning-samples#653]
Seems the training is not running so well. The accuracy even after 500 epochs is way too low. I also tried the full 3600 images data set  but not much better, so I have a feeling there’s something wrong here. 

I just cloned the repo and got these very low accuracy results with VS 2019 with .NET Core 2.1 or 3.0. 

Is anyone else seeing that with the sample or is it just me and maybe my machine config? 
https://github.com/dotnet/machinelearning-samples/tree/master/samples/csharp/getting-started/DeepLearning_ImageClassification_Training

See the super low accuracy here: 
![image](https://user-images.githubusercontent.com/7439129/65017822-e9255280-d927-11e9-9623-07d3ada3893f.png)

I’ve also tried InceptionV3 instead of ResNetv2101 as architecture but that raises an ArgumentOutOfTRangeException: 
![image](https://user-images.githubusercontent.com/7439129/65017839-f4787e00-d927-11e9-915c-24367154adf4.png)

**Comment by CESARDELATORRE** [comment]
We're not able to reproduce it and several people have tried this sample and get the right accuracy, around 90%. It must be something related to your environment, but it is weird..
Let's not close the issue until the cause has been isolated, ok?

**Comment by reneschulte** [comment]
Agreed. It's some weird edge case and only a few might run into it. Still worth checking as it could be surfacing a deeper, hidden issue just under the conditions of my machine. I will try another machine tomorrow. 

Any other ideas I should try?

**Comment by CESARDELATORRE** [comment]
Adding @codemzs he might have ideas on what to try to find out what's causing this issue.

**Comment by iangithub** [comment]
I tried it , but look very well
![image](https://user-images.githubusercontent.com/4374243/65097107-168e0100-d9f9-11e9-9d1b-a17462677ddf.png)

**Comment by reneschulte** [comment]
I've tried on another machine b) and get good results there. 

The machine a) that gets these bad results is Windows 10 Version 1903 (OS Build 1836.356). Also I have Python and TensorFlow-Base and TensorFlow-GPU 1.12 on that machine a) installed via Anaconda. Maybe there's some registry key it still sets that might mess it up using an old lib or so? 

Again, I think it would still make sense to narrow down the root cause for machine a) as others might experience similar.

**Comment by Fontijne** [comment]
Hello All,

I'm facing exactly the same problems.  Also after running the large data-set the accuracy is still very low.
Most of the predictions are wrong but the score is always 1.

I also tried the other architecture (InceptionV3) but I had the same error showing up.

My system is Windows 10 Pro (Build 18362)

Any suggestions are more than welcom.

**Comment by CESARDELATORRE** [comment]
@codemzs - Can you advice on this issue? There's something in the infrastructure/OS that might be causing these strange issues.. We're not able to repro.

**Comment by reneschulte** [comment]
Hi, I just tried again with latest from master ddbe84ff23943fdc528178a17dc823f5d4802705 and still it's failing to train properly I think. This is the result when I plug in the created/trained imageClassifier.zip model into the Predict program:

![image](https://user-images.githubusercontent.com/7439129/70049513-261af000-15cd-11ea-9705-792a0ceff905.png)

**Comment by FloP93** [comment]
I also have a similar problem.
The acuraccy is very low. 
![image](https://user-images.githubusercontent.com/34069654/70630645-2bf77d80-1c2c-11ea-9e60-ead8d4fc8d24.png)

And also if i try to run it on GPU with SciSharp.TensorFlow.Redist-Windows-GPU (1.14.1 or 1.14.0) i get the followin error:
![image](https://user-images.githubusercontent.com/34069654/70630781-6cef9200-1c2c-11ea-88f5-af40b3908faf.png)

My stats:
Windows 10
Visual Studio Enterprise 2017
8GB Ram
AMD Ryzen Processor
GeForce GTX 1070 Graphicscard

Hope that might help

**Comment by IveJ** [comment]
Hi freind,

Which loss func you implement.

**Comment by FloP93** [comment]
> Hi freind, Which loss func you implement.

I used the pretrained Networks MobilenetV2 and InceptionV3, which from my knowledge both use Relu in the hidden layers and Softmax as the last one. For the test i used the ImageClassification.Train sample : https://github.com/dotnet/machinelearning-samples/tree/master/samples/csharp/getting-started/DeepLearning_ImageClassification_Training/ImageClassification.Train

**Comment by reneschulte** [comment]
Same as @FloP93 I just used the pre-trained model with the sample. Nothing else.
